ac/ac-autocode/src/ActivityRunner.jsx

```diff
@@ -1,30 +1,91 @@
 // @flow
 
-import React from 'react';
+import React, { Component } from 'react';
 import type { ActivityRunnerT } from 'frog-utils';
 
-const ActivityRunner = ({ activityData }: ActivityRunnerT) => {
-  const submitCode = () => {};
+export default class ActivityRunner extends Component {
+  state: {
+    inputCode: string,
+    outputFeed: string
+  };
 
-  return (
-    <div>
-      <h1>{activityData.config.title || 'Coding exercise'}</h1>
+  constructor(props: ActivityRunnerT) {
+    super(props);
+
+    this.state = {
+      inputCode: 'print "Hello World"',
+      outputFeed: "You'll see the output of your code here"
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.runit = this.runit.bind(this);
+  }
+
+  componentDidMount() {
+    const script = document.createElement('script');
+
+    script.src = 'http://www.skulpt.org/static/skulpt.min.js';
+    script.async = false;
+    if (document.body != null) {
+      document.body.appendChild(script);
+    }
+
+    const script2 = document.createElement('script');
+    script2.src = 'http://www.skulpt.org/static/skulpt-stdlib.js';
+    script2.async = false;
+    if (document.body != null) {
+      document.body.appendChild(script2);
+    }
+  }
+
+  builtinRead(x: string) {
+    if (
+      window.Sk.builtinFiles === undefined ||
+      window.Sk.builtinFiles['files'][x] === undefined
+    )
+      throw "File not found: '" + x + "'";
+    return window.Sk.builtinFiles['files'][x];
+  }
+
+  outfunction(text: string) {
+    window.Sk.runner.state.outputFeed =
+      window.Sk.runner.state.outputFeed + '\n' + text;
+  }
+
+  runit: Function;
+  runit() {
+    if (window.Sk) {
+      window.Sk.runner = this;
```

Comment:
Outfunction has a different scope because it is called by Sk (so using this.state in outfunction does not work).
This workaround creates an explicit link between Sk and the AtivityRunner.